common/djangoapps/config_models/models.py

```diff
@@ -35,33 +37,69 @@ def save(self, *args, **kwargs):
         Clear the cached value when saving a new configuration entry
         """
         super(ConfigurationModel, self).save(*args, **kwargs)
-        cache.delete(self.cache_key_name())
+        cache.delete(self.cache_key_name(*[getattr(self, key) for key in self.KEY_FIELDS]))
 
     @classmethod
-    def cache_key_name(cls):
+    def cache_key_name(cls, *args):
```

Comment:
Using a dictionary that includes the key names here (`**kwargs`) might result in more readable client code and better error checking.  Otherwise, clients will have to provide the keys in the right order, and they won't be annotated with their meaning.  The same comment applies to the `current()` method.

The error checking code would be

    if list(cls.KEY_FIELDS) != kwargs.keys():
        raise TypeError(…)